### ISSUE TITLE
Support for element-wise minimum and maximum in modeling language

### ISSUE THREAD

**Issue by the-moliver** [slaue/genosolver#2]
Thank you for your work on this, it's very impressive. 
I'm trying to evaluate it to replace CVXPY in our framework but got a bit stuck by the lack of ability to use element-wise minimum and maximum functions in defining the conditions. I would like to be able to do thing like ```maximum(0, w1 - w2) <= n``` 
Is this possible to do?

**Comment by slaue** [comment]
Thank you for your feedback and your feature request.

Currently, it is not possible to use the element-wise maximum or minimum. I will have a look at it and try to incorporate it directly into our framework.

However, for the constraint that you provide, you could equivalently write ```0 <= n```  and ```w1 - w2 <= n``` since ```maximum(a, b) <= c```   is equivalent to ```a <= c``` and ```b <= c```. Please correct me if I misunderstood you here.

**Comment by the-moliver** [comment]
Thank you for the response. You're right, that may not have been the best example, but there are use cases I have in mind for element-wise maximums, like constraining the projection of the positive part of a vector w, onto another vector f as being less than a scalar n, ie `sum(maximum(0, w) * f) <= n` I assume all the elementwise operations supported by CVXPY should be possible in theory, no?

**Comment by slaue** [comment]
You are right, there should be support for element-wise maximum and minimum. I will put it on the todo list. Meanwhile, as I described above, you can replace maximum(a, b) with two constraints. This is how it would be done internally as well. Doing it by hand is more error-prone and less user-friendly, though.

So, your constraint would look something like
```0 <= v, w <= v, sum(v * f) <= n```
